src/asm/lexer.c

```diff
@@ -2399,28 +2399,44 @@ int yylex(void)
 	return token;
 }
 
-static char *startCapture(void)
+static void startCapture(struct CaptureBody *capture)
 {
 	assert(!lexerState->capturing);
 	lexerState->capturing = true;
 	lexerState->captureSize = 0;
 	lexerState->disableMacroArgs = true;
 	lexerState->disableInterpolation = true;
 
+	capture->lineNo = lexer_GetLineNo();
+
 	if (lexerState->isMmapped && !lexerState->expansions) {
-		return &lexerState->ptr[lexerState->offset];
+		capture->body = &lexerState->ptr[lexerState->offset];
 	} else {
 		lexerState->captureCapacity = 128; /* The initial size will be twice that */
 		assert(lexerState->captureBuf == NULL);
 		reallocCaptureBuf();
-		return NULL; // Indicate to retrieve the capture buffer when done capturing
+		capture->body = NULL; // Indicate to retrieve the capture buffer when done capturing
 	}
 }
 
+static void endCapture(struct CaptureBody *capture)
+{
+	// This being NULL means we're capturing from the capture buf, which is `realloc`'d during
+	// the whole capture process, and so MUST be retrieved at the end
+	if (!capture->body)
+		capture->body = lexerState->captureBuf;
+	capture->size = lexerState->captureSize;
+
+	lexerState->capturing = false;
+	lexerState->captureBuf = NULL;
+	lexerState->disableMacroArgs = false;
+	lexerState->disableInterpolation = false;
+	lexerState->atLineStart = false;
```

Comment:
I think this specific line should *not* be part of `endCapture`, as it's not tearing down something that was set up by `startCapture`.

I *think* that it being factor-able is more a byproduct of the uses made for captures than an intrinsic property, if you catch my drift.

Reply:
I see, that makes sense.